learn/using-the-openapi-tools.md

```diff
@@ -13,135 +13,159 @@ redirect_from:
-
-- [Mock Service from OpenAPI](#mock-service-from-openapi)
-- [Client Stub from OpenAPI](#client-stub-from-openapi)
-- [Service to OpenAPI Export](#service-to-openapi-export)
-- [Client Stub for Service](#client-stub-for-service)
+# Ballerina OpenAPI Tools
+
+OpenAPI Specification is a specification that creates a RESTFUL contract for APIs, detailing all of its resources 
+and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina
+ OpenAPI tooling will make it easy for users to start development of a service documented in the OpenAPI contract in
+ Ballerina by generating Ballerina service and client skeletons. The OpenAPI tools provide the following capabilities.
+ 
+ 1. Generate the Ballerina Service or Client code for a given OpenAPI definition. 
+ 2. Export the OpenAPI definition of a Ballerina service.
+ 3. Validate service implementation for a given OpenAPI Contract.
+    
+The `openapi` command in Ballerina is used for OpenAPI to Ballerina and Ballerina to OpenAPI code generation. 
+Code generation from OpenAPI to Ballerina can produce `ballerina service stubs` and `ballerina client stubs`.
+OpenAPI compiler plugin will allow you to validate a service implementation against a OpenAPI contract during compile time. 
+This plugin ensures the implementation of a service does not deviate from OpenAPI contract.
+
+- [OpenAPI to Ballerina](#openAPI-to-ballerina)
+    - [Generate Service and Client Stub from OpenAPI Contract](#generate-service-and-client-stub-from-openapi-contract)
+        - [Modes](#modes)
+- [Ballerina to OpenAPI](#ballerina-to-openAPI)    
+    - [Service to OpenAPI Export](#service-to-openapi-export)
 - [Samples](#samples)
-    - [Mock Service from OpenAPI Sample](#mock-service-from-openapi-sample)
-    - [Client Stub from OpenAPI Sample](#client-stub-from-openapi-sample)
-    - [OpenAPI from Service Sample](#openapi-from-service-sample)
-    - [Client Stub from Service Sample](#client-stub-from-service-sample)
+    - [Generate Service and Client Stub from OpenAPI](#generate-service-and-client-stub-from-openAPI)
+    - [Generate OpenAPI contract from Service](#generate-openAPI-contract-from-service)
+- [OpenAPI Validator Compiler Plugin](#openAPI-validator-compiler-plugin)
+     
+
+### OpenAPI to Ballerina
+#### Generate Service and Client stub from OpenAPI Contract
+
+```bash
+ballerina openapi   -i <openapi-contract> 
+               [--service-name: generated files name]
+               [--tags: tags list]
+               [--operations: operationsID list]
+               [--mode service|client ]
+               [(-o|--output): output file path]
+```
+Generates both the Ballerina service and Ballerina client stub for a given OpenAPI file.
 
-## Mock Service from OpenAPI
+This `-i <openapi-contract>` parameter of the command is mandatory. As an input, it will take the path to the OpenAPI contract file ie `my-api.yaml` or `my-api.json`. 
```

Comment:
```suggestion
This `-i <openapi-contract>` parameter of the command is mandatory. As an input, it will take the path to the OpenAPI contract file (i.e., `my-api.yaml` or `my-api.json`). 
```